### PR TITLE
Fix `Set#to_a(&)`

### LINKED ISSUE #14518: `Set#to_a&` works the same as to_a
The current setup of `Set#to_a&` has the following example in the docs: `# Set{1, 2, 3, 4, 5}.to_a { |i| i // 2 } # => [0, 1, 2]`, which doesn't seem about right since there should be 5 elements present, as well as looking at the method so doesn't it actually do anything with the accepted block:

```
def to_a(& : T -> U) : Array(U) forall U
  array = Array(U).new(size)
  @hash.each_key do |key|
    array << yield key
  end
  array
end
 ```

### PR DESCRIPTION
**Pull request by meatball133**
⚠️ This pr includes breaking changes ⚠️ 

resolves #14518

### CODE REVIEW

File: spec/std/set_spec.cr [hunk 1]
```diff
@@ -286,6 +286,10 @@ describe "Set" do
     Set{1, 2, 3}.to_a.should eq([1, 2, 3])
   end
 
+  it "does support giving a block to to_a"
+    Set{1, 2, 3}.to_a{|x| x + 1}.should eq([2, 3, 4])
```
Comment (devnote-dev) [inline spec/std/set_spec.cr:290]:
```suggestion
    Set{1, 2, 3}.to_a{ |x| x + 1 }.should eq([2, 3, 4])
```

**Review by devnote-dev — APPROVED** [review]

File: spec/std/set_spec.cr [hunk 2]
```diff
@@ -286,6 +286,10 @@ describe "Set" do
     Set{1, 2, 3}.to_a.should eq([1, 2, 3])
   end
 
+  it "does support giving a block to to_a" do
+    Set{1, 2, 3}.to_a{ |x| x + 1 }.should eq([2, 3, 4])
```
Comment (Sija) [inline spec/std/set_spec.cr:290]:
```suggestion
    Set{1, 2, 3}.to_a { |x| x + 1 }.should eq([2, 3, 4])
```

**Review by straight-shoota — APPROVED** [review]

**Comment by straight-shoota** [comment]
Technically it's breaking the current behaviour of this method. Of course it was bullocks, so the fix is necessary.
We don't categorize it as a breaking change but a bug fix. Still it'll be highlighted in the changelog due to the potential for breaking something.

**Review by beta-ziliani — APPROVED** [review]
Thanks @meatball133 !

**Pull request merged**